torchvision/prototype/features/_video.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+import warnings
+from typing import Any, cast, List, Optional, Tuple, Union
+
+import torch
+from torchvision.transforms.functional import InterpolationMode
+
+from ._feature import _Feature, FillTypeJIT
+from ._image import ColorSpace
+
+
+class Video(_Feature):
+    color_space: ColorSpace
+
+    def __new__(
+        cls,
+        data: Any,
+        *,
+        color_space: Optional[Union[ColorSpace, str]] = None,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[Union[torch.device, str, int]] = None,
+        requires_grad: bool = False,
+    ) -> Video:
+        data = torch.as_tensor(data, dtype=dtype, device=device)  # type: ignore[arg-type]
+        if data.ndim < 3:
+            raise ValueError
+        elif data.ndim == 3:
+            data = data.unsqueeze(0)
```

Comment:
I think for videos we can safely require exactly 4 dimensions.

Reply:
As in a video will always have shape `(F, C, H, W)` rather than arbitrary leading batch dimensions, i.e. `(*, F, C, H, W)`? If yes, what is the reason for this restriction given that all other features don't have this?

Or do you mean we can require the input of the constructor to always have *at least* 4 dimensions?

Reply:
I was suggesting the latter. Require the input of the constructor to always have at least 4 dimensions.

Reply:
@pmeier just curious what `F` stand for in `(F, C, H, W)` ? From what I know, usually the shape is either channel last, `(T, H, W, C)`, or channel first, `(C, T, H, W)`. Where `T` is `temporal` dimension.

Reply:
`F` here is the temporal dimension. I recommend using `T` in our comms to avoid misunderstandings.